tests.yaml

```diff
@@ -0,0 +1,239 @@
+# program. However, the depth of "tests" here stops at the boundary of test
+# programs which maintain their own subtests. I.e. do not add tests which
+# names are under control of such programs. You are encouraged to report them
+# to kcidb, but they're not recorded or regulated by this catalog. When adding
+# the test corresponding to a program, use its widely-known name, e.g. the
+# same one you would call it at a conference. If the program is only known
+# inside your company, prepend its name with the company name.
+#
+# The "home" property should point to the canonical, upstream location, and is
+# intended for identification and as a reference to more information on the
+# test, rather than a pointer to exact code being executed. The latter should
+# be described within test objects submitted to kcidb instead.
+#
+# E.g. the Linux Test Project test suite *should* be here under the
+# widely-known name "ltp". However, while its own tests, such as "chdir01" or
+# "clone03" are welcome to be reported to kcidb as "ltp.chdir01" or
+# "ltp.clone03", only the "ltp" test should be in this catalog. The LTP
+# project's home page or its main upstream repository should be referred to in
+# the "home" property, and not e.g. the specific code being executed, because
+# code versions will differ between test result origins and even executions.
+#
+# To decide where to place your test, follow organizational patterns already
+# in the file, and discuss any non-obvious cases when submitting the change.
+#
+# The objective of organization is to simplify locating tests as much as
+# possible, which can be helped by observing the following rules of thumb.
+#
+#   * Have only one organizational dimension per "tests" dictionary.
+#     E.g. if a "tests" dictionary has subsystem names as keys, then it
+#     shouldn't have device names there as well.
+#   * Avoid creating too many small categories. Instead, create a "misc"
+#     category and put all the miscellaneous tests there, together. Once the
+#     "misc" category accumulates enough certain tests for a category of their
+#     own, move them out.
+#   * Avoid placing programs together with categories in the same "tests"
+#     dictionary, if they could go into "misc" category instead.
+#
+# You can use the "kcidb-tests-validate" tool to check if this file is valid.
+# Like this:
+#
+#     kcidb-tests-validate < tests.yaml
+#
+# If you add the -u/--urls option, it will also check if all URLs are
+# reachable:
+#
+#     kcidb-tests-validate --urls < tests.yaml
+#
+# You can also use the "kcidb-tests-list" tool to print a flat list of known
+# test programs:
+#
+#   kcidb-tests-list < tests.yaml
+#
+title: The top-level "test" signifying testing in general
+description: |
+    Report results on this test when you don't have any more details on your
+    testing, and/or when you want to signal your testing for the build has
+    concluded.
+tests:
+    storage:
```

Comment:
What is the rationale for having such categories?  How about having a flat list of test suites instead, maybe with tags to group them if needed?

I fear having top-level categories might be a bit artificial and subjective, and if we enforce them in the test names we may be causing unnecessary complications.

Reply:
Yes, this is my fear as well. I agree this is subjective, and could be an unnecessary point of contention. I also think we can go with a flat list.

My only argument for this was that a flat list would be harder to page through when viewing results, and harder to navigate when cataloging new tests (although that's much less of a problem).

OTOH, we can have tags and even multiple categorization schemes outside of kcidb and use those when displaying results. However, I was hesitating to go with that, as the requisite queries/logic could become quite a bit complicated and would take a while to implement, and meanwhile we'd have to deal with a flat list.

Still, I don't feel strongly about this, and I think a flat list is a good way to go in general. We get less arguing when cataloging and agreeing on the names, and we can have as many categorization schemes as we want. So if you agree, and don't see the above arguments as a problem, let's go with a flat list :smiley:.

Reply:
One note: if we go with the flat list, it would be best to have test categorization/tagging data inside whatever database we'll use for storing test results. At least in addition to whatever place will be authoritative, if any. That would make it possible to organize the data using queries only and might simplify the code overall.

Reply:
> So if you agree, and don't see the above arguments as a problem, let's go with a flat list .

Yes I'm still in favour of a flat list, and I agree with what you wrote.

The tags should indeed be stored in the database in test records.  However, they need to come from somewhere, so having them in the test catalogue and using them when submitting the results would be one possibility.  If not in the test catalogue, it could also be in `test-configs.yaml` or an equivalent configuration file depending on how the YAML data ends up being shaped.